src/common/transformations/src/transformations/common_optimizations/fq_mul_fusion.cpp

```diff
@@ -116,7 +116,8 @@ ngraph::pass::FakeQuantizeMulFusion::FakeQuantizeMulFusion() {
             fq_node->input_value(2),
             get_adjusted_output_range(original_output_low),
             get_adjusted_output_range(original_output_high)});
-        if (transformation_callback(new_fq_node))
+        bool fq_on_weights = is_type<opset4::Constant>(data.get_node());
+        if (!fq_on_weights && transformation_callback(new_fq_node))
```

Comment:
1) Should FakeQuantizeMulFusion works only for constant subgraph (only for weights)? If only for constant subgraph, then condition has to be:
```
 if (!fq_on_weights || transformation_callback(new_fq_node))
```
Note, please, you can separate condition and check `fq_on_weights` before `new_fq_node` creation.

But if FakeQuantizeMulFusion should works only for constant subgraph (only for weights) then more logical way is update matcher: use `ngraph::pattern::wrap_type<opset1::Constant>()` instead `ngraph::pattern::any_input()` for zero input of `FakeQuantize`. In this case transformatiion will work faster. 

2) If FakeQuantizeMulFusion should works for not constant subgraph too, then very strange `transformation_callback` usage: `transformation_callback` is called only if some condition is satisfied. Is it expected?

Reply:
It can be for both weights and activations. Since FQ on weights is handled by LPT, there is no need to check if new_fq_node on weights is supported by the plugin.